### PR TITLE
Drop Core-JS polyfills

### PR DESCRIPTION
**Pull request by jridgewell**
Replaces it with a hand-written `Math.sign` and [my own](https://github.com/jridgewell/PJs) PJs Promise library.

### CODE REVIEW

File: third_party/pjs/promise.js [hunk 1]
```diff
@@ -0,0 +1,420 @@
+/**
+ * Constructs a ES6/Promises A+ Promise instance.
+ * @constructor
+ * @param {function(function(*=), function (*=))} resolver
+ */
+function Promise(resolver) {
```
Comment (cramforce) [inline third_party/pjs/promise.js]: Does this pass the A+ test suite? I didn't immediately see it in the repo.

File: third_party/pjs/promise.js [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+        }
+    };
+}
+
+/**
+ * Defers fn to another run loop.
+ *
+ * @type {function(function())}
+ */
+var nextTick = typeof setImmediate !== 'undefined' ? setImmediate
```
Comment (cramforce) [inline third_party/pjs/promise.js]: At least in AMP it is easy to do a polyfill based on postMessage to our own window.

Basically do

`window.addEventListener('message', flush)` and `window.postMessage('macro-task')`.

This is only relevant for older iOS and Android, but we want decent support there.

Comment (jridgewell) [inline third_party/pjs/promise.js]: Awesome, I'll add it.

File: third_party/pjs/promise.js (hunk 1 above)
Comment (jridgewell) [inline third_party/pjs/promise.js]: Yup, [it's compliant](https://github.com/jridgewell/PJs/blob/master/test/promises-spec.js).

File: third_party/pjs/promise.js (hunk 2 above)
Comment (jridgewell) [inline third_party/pjs/promise.js]: Updated.

File: third_party/pjs/promise.js [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+ * @param {function()} fn
+ */
+var defer = (function() {
+  /**
+   * Defers fn to another run loop.
+   */
+  var scheduleFlush = typeof setImmediate !== 'undefined' ? function() { setImmediate(flush); }
+    : window.postMessage ? (function() {
+      window.addEventListener('message', function(event) {
+        if (event.source === window && event.data === 'macro-task') {
```
Comment (cramforce) [inline third_party/pjs/promise.js]: There is really no need to check. All messages should be good times to flush.

File: 3p/environment.js [hunk 4]
```diff
@@ -255,9 +255,11 @@ function minTime(time) {
   return time;
 }
 
-listenParent('embed-state', function(data) {
-  inViewport = data.inViewport;
-  if (inViewport) {
-    becomeVisible();
-  }
-});
+window.installEmbedStateListener = function() {
```
Comment (cramforce) [inline 3p/environment.js]: You can just export this function. No need to marshall it through window.

File: third_party/pjs/promise.js (hunk 3 above)
Comment (jridgewell) [inline third_party/pjs/promise.js]: Fixed.

File: 3p/environment.js (hunk 4 above)
Comment (jridgewell) [inline 3p/environment.js]: Fixed.

**Comment by jridgewell** [comment]
It passed Travis! :confetti_ball: :confetti_ball: :confetti_ball:

**Comment by jridgewell** [comment]
Ping. Added unhandled Promise Rejection detection, too.

File: third_party/pjs/promise.js [hunk 5]
```diff
@@ -0,0 +1,438 @@
+/**
```
Comment (cramforce) [inline third_party/pjs/promise.js]: Just wondering: Could we install this from NPM instead?

**Comment by cramforce** [comment]
LGTM

Please separately do the unhandled error rejection timeout.

File: third_party/pjs/promise.js (hunk 5 above)
Comment (jridgewell) [inline third_party/pjs/promise.js]: I've never published it, but I could easily.

**Comment by jridgewell** [comment]
> Please separately do the unhandled error rejection timeout.

Separately?

**Comment by cramforce** [comment]
LGTM.

I'd prefer to do the dependency over NPM.

**Pull request merged**